carta/session.py

```diff
@@ -340,45 +340,65 @@ def cd(self, path):
+        make_active : {3}
+            Whether the image should be made active in the frontend. This only applies if an image is being appended. The default is ``True``.
         update_directory : {4}
             Whether the starting directory of the frontend file browser should be updated to the parent directory of the image. The default is ``False``.
         """
-        return Image.new(self, path, hdu, False, complex, expression, update_directory=update_directory)
+        directory, file_name = posixpath.split(path)
+        return Image.new(self, directory, file_name, hdu, append, False, make_active=make_active, update_directory=update_directory)
 
-    @validate(String(), String(r"\d*"), Boolean(), Constant(ArithmeticExpression), Boolean(), Boolean())
-    def append_image(self, path, hdu="", complex=False, expression=ArithmeticExpression.AMPLITUDE, make_active=True, update_directory=False):
-        """Append a new image, keeping any existing images.
+    @validate(String(), Constant(ComplexComponent), Boolean(), Boolean(), Boolean())
+    def open_complex_image(self, path, component=ComplexComponent.AMPLITUDE, append=False, make_active=True, update_directory=False):
+        """Open or append a new complex-valued image.
 
         Parameters
         ----------
         path : {0}
-            The path to the image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
-        hdu : {1}
-            The HDU to select inside the file.
-        complex : {2}
-            Whether the image is complex. Set to ``False`` by default.
-        expression : {3}
-            Arithmetic expression to use if appending a complex-valued image. The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
-        make_active : {4}
+            The path to the complex-valued image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        component : {1}
+            The complex component to select when opening the image. The default is :obj:`carta.constants.ComplexComponent.AMPLITUDE`.
+        append : {2}
+            Whether the image should be appended to existing images. By default this is ``False`` and any existing open images are closed.
+        make_active : {3}
+            Whether the image should be made active in the frontend. This only applies if an image is being appended. The default is ``True``.
+        update_directory : {4}
+            Whether the starting directory of the frontend file browser should be updated to the parent directory of the image. The default is ``False``.
+        """
+        directory, file_name = posixpath.split(path)
+        component = f'{component}("{file_name}")'
+        return Image.new(self, directory, component, "", append, True, make_active=make_active, update_directory=update_directory)
```

Comment:
Please change the name of the parameter that you are passing to `new` back to `expression`. The parameter passed into this function is an enum value which represents the component. In line 381 it is used to create an expression, and that is passed into the `new` function. The purpose of the name change was to highlight the difference between these two things.